### DISCUSSION TITLE
Consolidate information useState and handleSubmit (data)

### DISCUSSION THREAD

**Discussion by dajimene** [react-hook-form/react-hook-form#3195] [Q&A] [answered]
hello sorry my english is bad please your help, i'm initializing useState const [address, setAddress] = useState ({clientid: 1, type: 'HOME'}); In a component that has two entries email and name at the time of recording I get data from the entries name and email but I am missing the data of my initial state and send it as a single object to the backend as I do

**Comment by bluebill1049** [comment, marked as answer]
- provide a CSB
- you probably should use `reset()` API and set up `defaultValues` at `useForm`

**Reply by bluebill1049** [reply]
please provide a codesanbox, it's not readable as above.

**Reply by dajimene** [reply]
As I do for a single json object to load the imput + the value where the state starts.

https://codesandbox.io/s/peaceful-resonance-fmhnk?file=/src/App.js

**Reply by bluebill1049** [reply]
```
useForm({
  defaultValues: {
    idCliente: 'xxx'
  }
})
```
if async

```
useEffect(() => {
  reset(xxx);
}, [reset])
```
